frontend/src/components/labwork/step/LabworkStepOverviewPanel.tsx

```diff
@@ -39,7 +39,7 @@ const LabworkStepOverviewPanel = ({ stepID, refreshing, grouping, groupingValue,
 
 	const [isFetchingSamples, setIsFetchingSamples] = useState(false)
 	const [sampleAndLibraryList, setSampleAndLibraryList] = useState<SampleAndLibrary[]>([])
-  const displayedSamples = useAppSelector(selectLabworkStepsState).steps[stepID].displayedSamples
+  const displayedSamples = useAppSelector((state) => selectLabworkStepsState(state).steps[stepID].displayedSamples)
```

Comment:
isn't the useSelector used to prevent having to pass the state around all the time ?

Reply:
No. It's to access redux state. Generally you want your selector to be precise to minimize component rerendering since it uses the return value of the selector to check for changes in value.

Reply:
If you just have it return from selectLabworkStepsState, any changes under the entire LabworkStepsState will cause rerendering, including unrelated steps.

Reply:
oh i see it's like creating a new selector on the fly. Make you wonder what is gained using the selector in the first place though.

Reply:
To avoid rerendering. Maybe the cost of rerendering is higher than creating a new selector which is just a lambda?